ui/app/src/components/shared/ConfirmationModal.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <ModalView
+    :requestClose="requestClose"
+    :isOpen="isOpen"
+  > 
+    <div class="modal-inner">
+
+      <ConfirmationModalAsk 
+        v-if="state === 'confirming'"
+        @confirmed="$emit('confirmed')"
+        :title="title"
+        :confirmButtonText="confirmButtonText"
+      > 
+        <template v-slot:body>
+          <slot name="selecting"></slot>
+        </template>
+      </ConfirmationModalAsk>
+
+      <ConfirmationModalSigning
+        v-else
+        :state="state"
+        @closerequested="requestClose"
+        :transactionHash="transactionHash"
+        :transactionStateMsg="transactionStateMsg"
+      >
+
+        <template v-slot:signing>
+            <slot :name="!!this.$slots.signing ? 'signing' : 'common'"></slot> 
+        </template>
+
+        <template v-slot:confirmed>
+          <slot :name="!!this.$slots.signing ? 'confirmed' : 'common'"></slot>
+        </template>
+
+        <template v-slot:rejected>
+          <slot :name="!!this.$slots.signing ? 'rejected' : 'common'"></slot>
+        </template>
+
+        <template v-slot:failed>
+          <slot :name="!!this.$slots.signing ? 'failed' : 'common'"></slot>
+        </template>
+      </ConfirmationModalSigning>
+
+    </div>
+
+  </ModalView>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+import ModalView from "@/components/shared/ModalView.vue";
+import ConfirmationModalAsk from "@/components/shared/ConfirmationModalAsk.vue";
+import ConfirmationModalSigning from "@/components/shared/ConfirmationModalSigning.vue";
+
+export type ConfirmState =
+  | "selecting"
+  | "confirming"
+  | "signing"
+  | "confirmed"
+  | "rejected"
+  | "failed";
+
+export default defineComponent({
+  props: { 
+    isOpen: { type: Boolean, default: false }, 
```

Comment:
could this be redundant as it could be based on the state? Maybe you can pull that state logic in?

Reply:
That's a good point! I'll correct it now!